core/arch/arm/plat-imx/imx6_init.S

```diff
@@ -0,0 +1,342 @@
+	 * - NSec cannot access PLE (PLE bit16=0)
+	 * - NSec can use SIMD/VFP (CP10/CP11) (bit15:14=2b00, bit11:10=2b11)
+	 *
+	 * PCR = 0x00000001
+	 * - no change latency, enable clk gating
+	 */
+	movw r0, #0x4000
+	movt r0, #0x0000
+	write_sctlr r0
+
+	movw r0, #0x0041
+	movt r0, #0x0000
+	write_actlr r0
+
+	movw r0, #0x0C00
+	movt r0, #0x0002
+	write_nsacr r0
+
+	movw r0, #0x0000
+	movt r0, #0x0001
+	write_pcr r0
+
+	/*
+	 * GIC configuration
+	 *
+	 * per-CPU interrupts config are in accordance with GIC driver:
+	 *
+	 * ID0-ID7(SGI)   for Non-secure interrupts
+	 * ID8-ID15(SGI)  for Secure interrupts.
+	 * All PPI config as Non-secure interrupts.
+	 *
+	 * Register ICDISR0 = 0xFFFF00FF
+	 *
+	 * Register ICCPMR = 0x80
+	 */
+
+	ldr  r0, =GIC_DIST_BASE
+	mov  r1, #0xFFFF00FF
+	str  r1, [r0, #GIC_DIST_ISR0]
+
+	ldr  r0, =GIC_CPU_BASE
+	mov  r1, #0x80
+	str  r1, [r0, #CORE_ICC_ICCPMR]
+
+	mov pc, lr /* back to tzinit */
+UNWIND(	.fnend)
+END_FUNC plat_cpu_reset_early
+
+/*
+ * A9 secured config, needed only from a single core
+ *
+ * Use scratables registers R0-R3.
+ * No stack usage.
+ * LR store return address.
+ * Trap CPU in case of error.
+ *
+ * TODO: size optim in code
+ */
+FUNC plat_cpu_reset_late , :
+UNWIND(	.fnstart)
+
+	read_mpidr r0
+	ands r0, #3
+	beq _boot_late_primary_cpu
+
+_boot_late_secondary_cpu:
+	mov pc, lr
+
+_boot_late_primary_cpu:
+
+	/*
+	 * Snoop Control Unit configuration
+	 *
+	 * SCU is enabled with filtering off.
+	 * Both Secure/Unsecure can access SCU and timers
+	 *
+	 * 0x00 SCUControl = 0x00000060 !!!  should be 0x5 !
+	 * 0x04 SCUConfiguration =  ???
+	 * 0x0C SCUInvalidateAll (Secure cfg)
+	 * 0x40 FilteringStartAddress = 0x40000000
+	 * 0x44 FilteeringEndAddress - 0x80000000
+	 * 0x50 SCUAccessControl
+	 * 0x54 SCUSecureAccessControl
+	 */
+
+	/*
+	 * SCU Access Register : SAC = 0x0000000F
+	 * - both secure CPU access SCU
+	 */
+	ldr  r0, =SCU_BASE
+	movw r1, #0x000F
+	movt r1, #0x0000
+	str  r1, [r0, #SCU_SAC]
+
+	/*
+	 * SCU NonSecure Access Register : SNSAC : 0x00000FFF
+	 * - both nonsec cpu access SCU, private and global timer
+	 */
+	movw r1, #0x0FFF
+	movt r1, #0x0000
+	str  r1, [r0, #SCU_NSAC]
+
+	/*
+	 * Disallow NSec to mask FIQ [bit4: FW=0]
+	 * Allow NSec to manage Imprecise Abort [bit5: AW=1]
+	 * Imprecise Abort trapped to Abort Mode [bit3: EA=0]
+	 * In Sec world, FIQ trapped to FIQ Mode [bit2: FIQ=0]
+	 * IRQ always trapped to IRQ Mode [bit1: IRQ=0]
+	 * Secure World [bit0: NS=0]
+	 */
+	mov r0, #SCR_AW
+	write_scr r0		/* write Secure Configuration Register */
+
+	/* configure imx6 CSU */
+
+	/* first grant access of all peripherals to NS... */
+	ldr r0, =CSU_CSL_START
+	ldr r1, =CSU_CSL_END
+	ldr r2, =0xffffffff
+
+loop_csu:
+	str r2, [r0]
+	add r0, r0, #4
+	cmp r0, r1
+	bne loop_csu
+
+	/* then restrict key peripherals */
+
+	ldr r0, =CSU_CSL_5 /* POWER group */
+	ldr r2, =0x003300ff
+	str r2, [r0]
+
+	ldr r0, =CSU_CSL_16 /* TZASC1,TZASC2 */
+	ldr r2, =0x00330033
+	str r2, [r0]
+
```

Comment:
@yanyan-wrs 
The CSU configuration should be per platform from my view.  Since you named this file as imx6_init.S, this file may be shared by i.MX6SDL/SX/SL and etc. I prefer move the csu configuration to a per platform file, but not shared with i.MX6 family. What do you think?